bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -0,0 +1,165 @@
+
+/**
+ * The {@link OpenGarageHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+
+public class OpenGarageHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenGarageHandler.class);
+
+    private long refreshInterval;
+
+    private OpenGarageWebTargets webTargets;
+    private ScheduledFuture<?> pollFuture;
+
+    public OpenGarageHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            handleCommandInternal(channelUID, command);
+        } catch (OpenGarageCommunicationException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+        }
+    }
+
+    private void handleCommandInternal(ChannelUID channelUID, Command command) throws OpenGarageCommunicationException {
+        logger.warn("Received command {} for thing '{}' on channel {}", command, thing.getUID().getAsString(),
+                    channelUID.getId());
+        switch (channelUID.getId()) {
+            case OpenGarageBindingConstants.CHANNEL_OG_STATUS:
+                if (command instanceof OnOffType) {
+                    changeStatus(((OnOffType) command).equals(OnOffType.ON));
+                    return;
+                }
+                break;
+        }
+
+        if (RefreshType.REFRESH != command) {
+            logger.warn("Received command {} of wrong type for thing '{}' on channel {}", command, thing.getUID().getAsString(),
+                    channelUID.getId());
+        } 
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Opengarage Unit");
+        OpenGarageConfiguration config = getConfigAs(OpenGarageConfiguration.class);
+        logger.debug("config.host = {}, refresh = {}, password = {}, port = {}", config.host, config.refresh, config.password, config.port);
+        if (config.host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host address must be set");
+        } else {
+            webTargets = new OpenGarageWebTargets(config.host, config.port, config.password);
+            refreshInterval = config.refresh;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 500ms out, then every {} s", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void stopPoll() {
```

Comment:
why do you use synchronized here?

Reply:
I used this as parts of this binding were based on the Daikin binding.

Reply:
I think you can safely remove that.